File: test/config/USSDConfig.ts
/** USSD constants —  USSDConfig */

export const BALANCE_CHECK_CODE = '*199#';
export const USSD_DIAL_TIMEOUT = 30_000;
export const USSD_RESPONSE_TIMEOUT = 45_000;
export const USSD_COMPLETION_TIMEOUT = 60_000;
export const BALANCE_UPDATE_WAIT = 8_000;

export const COMPLETION_INDICATORS = [
  'main balance',
  'balance',
  'valid till',
  'validity',
  'rs.',
  '₹'
];

export const ERROR_INDICATORS = [
  'connection problem',
  'not available',
  'failed',
  'error',
  'try again'
];

export const USSD_APP_PACKAGES = [
  'com.android.phone',
  'com.google.android.dialer',
  'com.android.stk'
];

File: src/main/java/com/telecom/config/USSDConfig.java
package com.telecom.config;

/**
 * ✅ USSD CONFIGURATION
 */
public class USSDConfig {
    
    // USSD Codes
    public static final String BALANCE_CHECK_CODE = "*199#";
//    public static final String DATA_BALANCE_CODE = "*121#";
//    public static final String VALIDITY_CODE = "*199*2#";
//    
    // Timeouts (in milliseconds)
    public static final long USSD_DIAL_TIMEOUT = 5000;
    public static final long USSD_RESPONSE_TIMEOUT = 15000;
    public static final long USSD_COMPLETION_TIMEOUT = 20000;
    public static final long BALANCE_UPDATE_WAIT = 5000;
    
    // Completion Indicators
    public static final String[] COMPLETION_INDICATORS = {
        "USSD Check Completed",
        "Check Completed",
        "Balance:",
        "MSISDN:",
        "Validity:",
        "Main Balance"
    };
    
    // Error Indicators
    public static final String[] ERROR_INDICATORS = {
        "Error",
        "Failed",
        "Not supported",
        "Service not available",
        "Unable to process",
        "Invalid"
    };
    
    // App packages to force stop
    public static final String[] USSD_APP_PACKAGES = {
        "com.android.phone",
        "com.android.dialer",
        "com.google.android.dialer",
        "com.android.incallui"
    };
}